### PR TITLE
Checkconfig: Add validation for tide context policy

### PR DESCRIPTION
**Pull request by alvaroaleman**
/assign @fejta @stevekuznetsov 

To avoid discovering this at runtime with errors like
```
{"branch":"release-4.3","component":"tide","controller":"status-update","error":"failed to set up context register: contexts ci/prow/shellcheck, ci/prow/tf-lint, ci/prow/yaml-lint are defined as required and required if present","file":"prow/tide/status.go:371","func":"k8s.io/test-infra/prow/tide.(*statusController).setStatuses.func1","level":"error","msg":"getting expected status","org":"openshift","pr":3460,"repo":"installer","sha":"e05a6363aa9e40cd3f341ada8ee3beb9dcabeda3","time":"2020-04-22T15:18:55Z"}
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
> What happens when this processes regex in the branch list?

We just call `cfg.GetTideContextPolicy` here which is the same as what tide does.

File: prow/cmd/checkconfig/main.go [hunk 1]
```diff
@@ -929,3 +936,49 @@ func validateInRepoConfig(cfg *config.Config, filePath, repoIdentifier string) e
...
+	// attempt to clone the repo if its enabled
+	originalInRepoConfig := cfg.InRepoConfig
+	cfg.InRepoConfig = config.InRepoConfig{}
+	defer func() { cfg.InRepoConfig = originalInRepoConfig }()
+
+	var errs []error
+	for orgRepo, branches := range allKnownOrgRepoBranches {
+		split := strings.Split(orgRepo, "/")
+		if n := len(split); n != 2 {
+			errs = append(errs, fmt.Errorf("expected to get two results when splitting string %s by '/', got %d", orgRepo, n))
```
Comment (cjwagner) [inline prow/cmd/checkconfig/main.go:965]: This assumes that keys to the `presubmits:` section of Prow config are all `org/repo` format which is not true for Gerrit repos. e.g. https://github.com/GoogleCloudPlatform/oss-test-infra/blob/5ddad3c6a30bf826e36c1a9723a71ff7aea80f00/prow/prowjobs/kunit-review.googlesource.com/linux/kunit-linux-config.yaml#L2

This is blocking merge for that entire instance and is a regression so I'm going to revert this PR.

Comment (cjwagner) [reply prow/cmd/checkconfig/main.go:965]: https://github.com/kubernetes/test-infra/pull/17467

**Pull request merged**